client/lib/transformer.dart

```diff
@@ -154,3 +162,5 @@ const _ANALYTICS_CODE = '''<script>
   ga('send', 'pageview');
 
 </script>''';
+
+const _SURVEY_CODE = '<script async="" defer="" src="//survey.g.doubleclick.net/async_survey?site=4vq7qmqpfn4zjduve64c4h6lqa"></script>';
```

Comment:
So what's happening with the analytics is that the transformer looks for a
file at _CONFIG_PATH, which is lib/config/config.yaml. That's not present
in the checking, there's a sample version of it, and it's in the .gitignore
so it doesn't get checked in by accident. It's there in my copy that I use
to actually update the site. (It's a very robust deployment process)

If it's there, then that sets the config variable. The verifier and
analytics plugin check that for their values, and only do something if it's
present.

So we could use the same file for survey information, or add a different
one, put our specific survey number in there, and only replace the code if
it's present.


On Wed Feb 25 2015 at 11:28:52 AM Keerti Parthasarathy <
notifications@github.com> wrote:

> In client/lib/transformer.dart
> <https://github.com/dart-lang/dartdoc-viewer/pull/124#discussion_r25370316>
> :
>
> > @@ -154,3 +162,5 @@ const _ANALYTICS_CODE = '''<script>
> >    ga('send', 'pageview');
> >
> >  </script>''';
> > +
> > +const _SURVEY_CODE = '<script async="" defer="" src="//survey.g.doubleclick.net/async_survey?site=4vq7qmqpfn4zjduve64c4h6lqa"></script>';
>
> So do we need to add a check elsewhere?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dart-lang/dartdoc-viewer/pull/124/files#r25370316>.
>